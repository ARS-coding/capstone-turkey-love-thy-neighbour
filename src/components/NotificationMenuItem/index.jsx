import React from "react";

import { CloseButton } from "react-bootstrap";

import { useSelector } from "react-redux";

import firebaseApp, { firestore } from "../../firebaseConfig";

import "./index.scss";

function NotificationMenuItem({ invitationNotificationObject }) {
  const { message, id } = invitationNotificationObject;
  const uid = useSelector((state) => state.user.authCred?.uid);

  function handleButtonClick(event) {
    const buttonText = event.target.innerText;

    firestore
      .collection("users")
      .doc(uid)
      .update({
        // eslint-disable-next-line import/no-named-as-default-member
        invitationNotifications: firebaseApp.firestore.FieldValue.arrayRemove({
          message,
          id,
        }),
      });

    if (buttonText === "Good") {
      firestore
        .collection("data")
        .doc("feedbackData")
        .update({
          // eslint-disable-next-line import/no-named-as-default-member
          positiveFeedback: firebaseApp.firestore.FieldValue.increment(1),
        });
    }
    if (buttonText === "Bad") {
      firestore
        .collection("data")
        .doc("feedbackData")
        .update({
          // eslint-disable-next-line import/no-named-as-default-member
          negativeFeedback: firebaseApp.firestore.FieldValue.increment(1),
        });
    }
  }

  return (
    <div className="notification-menu-item">
      <div className="notification-menu-item-content">
        <p className="text-center">{message}</p>
        <div className="feedback-buttons d-flex justify-content-center">
          <button
            type="button"
            className="me-3"
            id={id}
            onClick={handleButtonClick}
          >
            Good
          </button>
          <button type="button" id={id} onClick={handleButtonClick}>
            Bad
          </button>
          <CloseButton onClick={handleButtonClick} />
        </div>
      </div>
    </div>
  );
}

export default NotificationMenuItem;
