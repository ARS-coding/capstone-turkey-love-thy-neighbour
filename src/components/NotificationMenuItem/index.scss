div.notification-menu-item {
  border-top: 1px solid lightgray;
  border-bottom: 1px solid lightgray;
  margin: 0 0 7px 0;
  position: relative;

  & > div.notification-menu-item-content {
    & > p {
      margin: 15px 5px 7px 5px;
    }

    & > div.feedback-buttons {
      margin-bottom: 7px;

      button:last-child {
        position: absolute;
        top: 0;
        right: 0;
        background-color: transparent;
        border: none;
        font-size: 10px;
      }
    }
  }
}
