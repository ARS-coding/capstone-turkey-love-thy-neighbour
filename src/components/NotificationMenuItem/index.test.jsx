import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import NotificationMenuItem from ".";

describe("NotificationMenuItem", () => {
  const data = {
    invitationNotificationObject: {
      message: "test",
      id: "test",
    },
  };
  const mockState = {
    user: {
      authCred: {
        uid: "Test Uid",
      },
    },
  };
  const mockStore = configureStore();
  let store;
  let tree;

  beforeEach(() => {
    store = mockStore(mockState);
    tree = renderer
      .create(
        <Provider store={store}>
          <NotificationMenuItem
            invitationNotificationObject={data.invitationNotificationObject}
          />
        </Provider>
      )
      .toJSON();
  });

  it("renders NotificationMenuItem correctly", () => {
    expect(tree).toMatchSnapshot();
  });
});
