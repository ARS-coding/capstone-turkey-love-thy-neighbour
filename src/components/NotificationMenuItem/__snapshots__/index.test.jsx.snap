// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationMenuItem renders NotificationMenuItem correctly 1`] = `
<div
  className="notification-menu-item"
>
  <div
    className="notification-menu-item-content"
  >
    <p
      className="text-center"
    >
      test
    </p>
    <div
      className="feedback-buttons d-flex justify-content-center"
    >
      <button
        className="sc-bdnxRM sc-jrsJWt hPJaof kTNuYI me-3"
        id="test"
        onClick={[Function]}
        type="button"
      >
        👍🏼
      </button>
      <button
        className="sc-bdnxRM sc-kEqXSa hPJaof lcbPMi"
        id="test"
        onClick={[Function]}
        type="button"
      >
        👎🏼
      </button>
      <button
        aria-label="Close"
        className="btn-close"
        onClick={[Function]}
        type="button"
      />
    </div>
  </div>
</div>
`;
